src/tests/unittests/test_data.py

```diff
@@ -154,3 +132,355 @@ def test_load_dataset(verbose=0):
+        p = ManualParameter('dummy')
+        x = p[0:10:1]
+        yname = 'measured'
+        y = [np.arange(len(x)).reshape((len(x))), np.arange(len(x)).reshape((len(x)))]
+        self.assertRaisesRegex(ValueError, 'The number of measurement names does not match the number of measurements',
+                               qtt.data.makeDataSet1D, x, yname, y, return_names=False)
+
+    def test_makedataset1d_shape_measuredata_nok(self):
+        p = ManualParameter('dummy')
+        x = p[0:10:1]
+        y = [np.arange(len(x)+1)]
+
+        self.assertRaisesRegex(ValueError,
+                               'Measured data must be a sequence with shape matching the setpoint arrays shape',
+                               qtt.data.makeDataSet1D, x, 'y', y, return_names=False)
+
+    def test_makedataset1d_shape_measuredata_second_nok(self):
+        p = ManualParameter('dummy')
+        x = p[0:10:1]
+        y = [np.arange(len(x)), np.arange(len(x)+1)]
+
+        self.assertRaisesRegex(ValueError,
+                               'Measured data must be a sequence with shape matching the setpoint arrays shape',
+                               qtt.data.makeDataSet1D, x,
+                               ['y1', 'y2'], y, return_names=False)
+
+    def test_dataset_labels_dataset2dplain(self):
+        v = [0]
+        h = [0]
+        measurement = np.arange(len(v) * len(h)).reshape((len(v), len(h)))
+
+        ds = qtt.data.makeDataSet2Dplain('horizontal', h, 'vertical', v, 'z', measurement,
+                                         xunit='mV', yunit='Hz', zunit='A')
+
+        self.assertEqual(qtt.data.dataset_labels(ds), 'z')
+        self.assertEqual(qtt.data.dataset_labels(ds, 'x'), 'horizontal')
+        self.assertEqual(qtt.data.dataset_labels(ds, 1), 'horizontal')
+        self.assertEqual(qtt.data.dataset_labels(ds, 'y'), 'vertical')
+        self.assertEqual(qtt.data.dataset_labels(ds, 'y', add_unit='True'), 'vertical [Hz]')
+
+    def test_makedataset2dplain_type_measurement_names_nok(self):
+        v = [0, 1, 2.]
+        h = [0., 1, 2, 3, 4]
+        measurement = np.arange(len(v) * len(h)).reshape((len(v), len(h)))
+        z = [measurement, measurement]
+        self.assertRaisesRegex(TypeError, 'Type of measurement names must be str or qcodes.Parameter',
+                               qtt.data.makeDataSet2Dplain, 'horizontal', h, 'vertical', v, ['z1', 2],
+                               z, xunit='mV', yunit='Hz', zunit='A')
+
+    def test_dataset2dplain_shape_measuredata_list_ok(self):
+        v = [0, 1, 2.]
+        h = [0., 1, 2, 3, 4]
+        measurement = np.arange(len(v) * len(h)).reshape((len(v), len(h)))
+        z = [measurement, measurement]
+        data_set = qtt.data.makeDataSet2Dplain('horizontal', h, 'vertical', v, ['z1', 'z2'],
+                                               z, xunit='mV', yunit='Hz', zunit='A')
+        self.assertTrue(data_set.z1.shape == np.ones((len(v), len(h))).shape)
```

Comment:
General comment about testing equality of Numpy arrays. Is it sufficient to only check whether the shapes of the arrays are equal? I always check via:
```
desired_array = ???
self.assertTrue(np.array_equal(data_set.z1, desired_array))
```

Reply:
I changed where possible to the broader test with use of np.array_equal